python/tests/test_combinatorics.py

```diff
@@ -481,6 +481,31 @@ def test_is_symmetrical(self):
         six_leaf_sym = RankTree(children=[three_leaf_asym, three_leaf_asym])
         assert six_leaf_sym.is_symmetrical()
 
+    @pytest.mark.parametrize("n", range(7))
+    def test_leaves(self, n):
+        for tree in RankTree.all_unlabelled_trees(n):
+            first_labelled_tree = RankTree.unrank(n, (tree.shape_rank(), 0))
+            assert list(first_labelled_tree.leaves()) == list(range(n))
```

Comment:
This test would pass if the leaves() iterator was implemented as ``return range(n)``. I don't think there's much value in iterating over the many, many trees here, it's just wasting electricity.

A much better test would be a few small hand rolled examples. Confirming we get the right result on a handful of chosen examples would be much better here.